From: Bart Van Assche <bvanassche@acm.org>
Origin: Upstream, https://sourceforge.net/p/net-snmp/code/ci/a0df31c18c513a0d79f4d526b1af7fad48748e57/
Bug: https://sourceforge.net/p/net-snmp/bugs/2968/
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1835818
Date: Fri, 26 Jul 2019 21:40:12 -0700
Subject: [PATCH] HOST-MIB: Fix a recently introduced bug

Fixes: cf41e6e91015 ("HOST-MIB: Skip autofs entries")

Note: this bug was not introduced by Josef but by me.
---
 agent/mibgroup/host/hrh_storage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: net-snmp-5.7.3+dfsg/agent/mibgroup/host/hrh_storage.c
===================================================================
--- net-snmp-5.7.3+dfsg.orig/agent/mibgroup/host/hrh_storage.c
+++ net-snmp-5.7.3+dfsg/agent/mibgroup/host/hrh_storage.c
@@ -372,7 +372,7 @@ really_try_next:
                                    NETSNMP_DS_AGENT_SKIPNFSINHOSTRESOURCES) &&
             Check_HR_FileSys_NFS())
             return NULL;
-        if (Check_HR_FileSys_AutoFs())
+        if (HRFS_entry && Check_HR_FileSys_AutoFs())
             return NULL;
         if (store_idx <= NETSNMP_MEM_TYPE_MAX ) {
 	    mem = (netsnmp_memory_info*)ptr;
